integrated_channels/moodle/client.py

```diff
@@ -311,38 +312,191 @@ def _wrapped_create_course_completion(self, user_id, payload):
         return self._post(params)
 
     @moodle_request_wrapper
-    def create_content_metadata(self, serialized_data):
+    def _wrapped_create_content_metadata(self, payload):
+        return self._post(payload)
+
+    @moodle_request_wrapper
+    def _wrapped_update_content_metadata(self, payload):
+        return self._post(payload)
+
+    @moodle_request_wrapper
+    def _get_course_forum(self, course_id):
         """
-        The below assumes the data is dict/object.
-        Format should look like:
-        {
-          courses[0][shortname]: 'value',
-          courses[0][fullname]: 'value',
-          [...]
-          courses[1][shortname]: 'value',
-          courses[1][fullname]: 'value',
-          [...]
+        Obtains a list of forums for a given course id.
+        """
+        params = {
+            'wsfunction': 'mod_forum_get_forums_by_courses',
+            'courseids[]': course_id,
         }
+        return self._post(params)
+
+    def get_announcement_forum(self, course_id):
         """
-        serialized_data['wsfunction'] = 'core_course_create_courses'
-        response = self._post(serialized_data)
-        return response
+        The response is a list type, because of course it is.
+        For new courses, there should only be 1 forum and it should be Announcements
+        """
+        response = self._get_course_forum(course_id)
+        return response[0]['id']
```

Comment:
Fun fact that I'll probably want to address with Ken: if you make the course format single activity, you cannot add other forums. So in our flow, we'd have 0-1 forums (and 0 throws an exception as I have unfortunately found out already, which is why the course format change was added later).

So I agree in general that we should conditionalize on the Announcements forum specifically, however with the state of things it's not necessary. Even if users change the format, unfortunately during updates it will be changed back (actually unsure how users messing with settings is going to mess everything up - another question for Ken!)

Still it shouldn't be particularly difficult to check the subject(? I forget what the response object here looks like) of the forum and isolate on that, so I will check into that.

Reply:
Testing different settings seems like it's gonna be a time and a half 😬